src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/BuildTimeController.cs

```diff
@@ -0,0 +1,128 @@
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.Kusto;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Data;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    /// <summary>
+    ///   Exposes methods to Read <see cref="BuildTime"/>s
+    /// </summary>
+    [Route("buildtime")]
+    [ApiVersion("2019-01-16")]
+    public class BuildTimeController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        private readonly IKustoClientProvider _kustoClientProvider;
+
+        public BuildTimeController(
+            BuildAssetRegistryContext context,
+            IKustoClientProvider kustoClientProvider)
+        {
+            _context = context;
+            _kustoClientProvider = kustoClientProvider;
+        }
+
+        /// <summary>
+        /// Gets the average official build time and average pr build time for a given default channel
+        /// This is captured for generating the longest build times for the dependency flow graph
+        /// </summary>
+        /// <param name="id">Default channel id</param>
+        [HttpGet("{id}")]
+        [SwaggerApiResponse(HttpStatusCode.OK, Type = typeof(BuildTime), Description = "Gets the average official build time and average pr build time for a given default channel")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> GetBuildTimes([Required]int id)
+        {
+            Data.Models.DefaultChannel defaultChannel = await _context.DefaultChannels.FindAsync(id);
+
+            if (defaultChannel == null)
+            {
+                return NotFound();
+            }
+
+            var properties = new ClientRequestProperties();
+
+            // There is a mismatch between how we store the names of the repositories in BAR
+            // and how we store them in Kusto. Normalize to the last part of the url.
+            properties.SetParameter("_Repository", defaultChannel.Repository.Split("/").Last());
+            properties.SetParameter("_SourceBranch", defaultChannel.Branch);
+
+            string queryParameters = String.Join(",", properties.Parameters.Select(p => $"{p.Key}:{KustoDataTypes.String}"));
+
+            // We only care about builds that complete successfully or partially successfully 
+            // from the given repository. We summarize duration of the builds over the last 7 days.
+            // There are multiple different definitions that run in parallel, so we summarize
+            // on the definition id and ultimately choose the definition that took the longest.
+            string commonQueryText = @"| where Repository endswith _Repository
+                | where Result != 'failed' and Result != 'canceled' 
+                | where FinishTime > ago(7d) 
```

Comment:
I've considered making the number of days we are summarizing over user specified so that someone could gather data over a longer or shorter range. Thoughts @mmitche?

Reply:
That makes sense as an optional query parameter.